File: Lab/Atv-1/A_Recursivo.java
import java.util.Scanner;

public class A_Recursivo {
    public static boolean isUppercase(char c) {
        return (c >= 'A' && c <= 'Z');
    }

    public static boolean isEnd(String str) {
        return str.equals("FIM");
    }

    public static int countUppercase(String str, int position) {
        if (position >= str.length()) {
            return 0;
        }
        if (isUppercase(str.charAt(position))) {
            return 1 + countUppercase(str, position + 1);
        }
        return countUppercase(str, position + 1);
    }

    public static int countUppercase(String str) {
        return countUppercase(str, 0);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String str;

        do {
            str = scanner.nextLine();
            if (!isEnd(str)) {
                int count = countUppercase(str);
                System.out.println(count);
            }
        } while (!isEnd(str));
    }
}

File: Lab/Atv-1/A_Recursivo.c
#include <stdio.h>
#include <string.h>

int isUppercase(char c)
{
      return (c >= 'A' && c <= 'Z');
}

int isEnd(const char *str)
{
      return strcmp(str, "FIM") == 0;
}

int countUppercaseRecursive(const char *str, int index)
{
      if (str[index] == '\0')
      {
            return 0;
      }

      int count = 0;
      if (isUppercase(str[index]))
      {
            count = 1;
      }

      return count + countUppercaseRecursive(str, index + 1);
}

int main()
{
      char str[100];
      int count;

      do
      {
            fgets(str, sizeof(str), stdin);
            strtok(str, "\n");
            if (!isEnd(str))
            {
                  count = countUppercaseRecursive(str, 0);
                  printf("%d\n", count);
            }
      } while (!isEnd(str));

      return 0;
}